### PR TITLE
Hot 2510 dob cannot be future date

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
### Jira Story

- [Date of birth cannot be a future date rule HOT-2510](https://osi-cwds.atlassian.net/browse/HOT-2510)

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/selectors/screening/personShowSelectors.js [hunk 1]
```diff
@@ -110,6 +115,11 @@ const csecTypesSelector = (state, personId) => {
   )
 }
 
+const selectDobError = (person) => combineCompact(
```
Comment (wmitchell-cambria) [inline app/javascript/selectors/screening/personShowSelectors.js]: You didn't happen to stumble across any ideas for how we might combine these into one location, did you? I hate that we have to define our show mode and edit mode validations in two places, because they are almost identical functions every time.

File: app/javascript/selectors/screening/personShowSelectors.js [hunk 2]
```diff
@@ -177,13 +187,13 @@ export const getFormattedPersonInformationSelector = (state, personId) => {
     CSECTypes: {value: csecTypesSelector(state, personId), errors: []},
     csecStartedAt: {value: (person.getIn(['csec', 0, 'start_date']) && dateFormatter(person.getIn(['csec', 0, 'start_date']))), errors: []},
     csecEndedAt: person.getIn(['csec', 0, 'end_date']) && dateFormatter(person.getIn(['csec', 0, 'end_date'])),
-    dateOfBirth: dateOfBirth,
-    ethnicity: getEthnicity(person),
+    dateOfBirth: {value: person.get('date_of_birth') && dateFormatter(person.get('date_of_birth'))},
```
Comment (wmitchell-cambria) [inline app/javascript/selectors/screening/personShowSelectors.js]: I also hate how we use `{value, errors}` format here and `dateOfBirth, dateOfBirthErrors` in the form. Again, totally not your fault, and we should just keep an eye out for ways to improve this going forward. It makes adding validations more difficult and error prone, which feels like risky tech debt approaching Hotline 1.0.

File: app/javascript/views/people/PersonDemographicsForm.jsx [hunk 3]
```diff
@@ -97,6 +98,7 @@ PersonDemographicsForm.propTypes = {
   approximateAgeUnit: PropTypes.string,
   dateOfBirth: PropTypes.string,
   dateOfBirthIsRequired: PropTypes.bool,
+  dobError: PropTypes.array,
```
Comment (wmitchell-cambria) [inline app/javascript/views/people/PersonDemographicsForm.jsx]: nit: The abbreviation feels out of place here. Note that `dateOfBirthError` is shorter than `dateOfBirthIsRequired`, which is an already existing prop.

File: spec/javascripts/selectors/screening/personDemographicsFormSelectorsSpec.js [hunk 4]
```diff
@@ -57,6 +59,14 @@ describe('personDemographicFormSelectors', () => {
       expect(getPersonDemographicsSelector(state, '1').get('dateOfBirth')).toEqual('13/0/-514')
     })
 
+    it('returns an error if date is in the future', () => {
```
Comment (wmitchell-cambria) [inline spec/javascripts/selectors/screening/personDemographicsFormSelectorsSpec.js]: If you have a moment, can you add a unit test for when the date is today? Ferb validation ran into an off-by-one issue where today and yesterday were counted as the future, I believe. Maybe I'm being paranoid.

File: spec/javascripts/selectors/screening/personShowSelectorsSpec.js [hunk 5]
```diff
@@ -422,6 +439,24 @@ describe('personShowSelectors', () => {
       })
     })
 
+    describe('dateOfBirth', () => {
+      it('returns no error if date is current or in the past', () => {
+        const today = moment().toISOString()
+        const people = [{id: 'one', date_of_birth: today}]
```
Comment (wmitchell-cambria) [inline spec/javascripts/selectors/screening/personShowSelectorsSpec.js]: Nice. :-)

File: app/javascript/selectors/screening/personDemographicsFormSelectors.js [hunk 6]
```diff
@@ -25,6 +31,7 @@ export const getPersonDemographicsSelector = (state, personId) => {
     approximateAgeUnit: person.getIn(['approximate_age_units', 'value']) || '',
     dateOfBirth: person.getIn(['date_of_birth', 'value']) || '',
     dateOfBirthIsRequired: roles.includes('Victim'),
+    dobError: selectDobError(person),
```
Comment (wmitchell-cambria) [inline app/javascript/selectors/screening/personDemographicsFormSelectors.js]: Do we display these errors all the time, or only once the field is touched? I guess all the time is OK in this case, since we should not be receiving a future date from the back-end, so the only way it can happen is for the user to add it.

File: app/javascript/views/people/PersonDemographicsForm.jsx (hunk 3 above)
Comment (NagaPradeepGajula-osi) [reply app/javascript/views/people/PersonDemographicsForm.jsx]: Alright I will get the abbreviation done.

File: spec/javascripts/selectors/screening/personDemographicsFormSelectorsSpec.js (hunk 4 above)
Comment (NagaPradeepGajula-osi) [reply spec/javascripts/selectors/screening/personDemographicsFormSelectorsSpec.js]: sure not an issue.

File: spec/javascripts/selectors/screening/personShowSelectorsSpec.js (hunk 5 above)
Comment (NagaPradeepGajula-osi) [reply spec/javascripts/selectors/screening/personShowSelectorsSpec.js]: 👍

File: app/javascript/selectors/screening/personDemographicsFormSelectors.js (hunk 6 above)
Comment (NagaPradeepGajula-osi) [reply app/javascript/selectors/screening/personDemographicsFormSelectors.js]: This is all the time.

File: app/javascript/selectors/screening/personShowSelectors.js (hunk 2 above)
Comment (NagaPradeepGajula-osi) [reply app/javascript/selectors/screening/personShowSelectors.js]: Could have refactored by any new means, but I didn't want to have mixture of different validations. As you mentioned we should get this via tech debt though it's risky.
